src/pages/config/config-details.js

```diff
@@ -1,74 +1,76 @@
-import { useEffect, useState } from "react";
+import { useEffect, useState, useMemo, useCallback } from "react";
 import toast from "react-hot-toast";
-import { useNavigate, useParams } from "react-router-dom";
 import {
-  decodeUrlSearchParams,
-  updateParams
-} from "../../components/Canary/url";
-
-import { SearchLayout } from "../../components/Layout";
+  useNavigate,
+  useParams,
+  useSearchParams,
+  useOutletContext
+} from "react-router-dom";
 import { toastError } from "../../components/Toast/toast";
 import { Modal } from "../../components/Modal";
 import { IncidentCreate } from "../../components/Incidents/IncidentCreate";
 import { getConfig } from "../../api/services/configs";
 import { Loading } from "../../components/Loading";
+import { JSONViewer } from "../../components/JSONViewer";
+import { BreadcrumbNav } from "../../components/BreadcrumbNav";
+import { Button } from "../../components/Button";
 
 export function ConfigDetailsPage() {
   const navigate = useNavigate();
   const { id } = useParams();
+  const [params, setParams] = useSearchParams();
   const [isLoading, setIsLoading] = useState(true);
   const [showIncidentModal, setShowIncidentModal] = useState(false);
   const [checked, setChecked] = useState({});
   const [configDetails, setConfigDetails] = useState();
-  const [jsonLines, setJsonLines] = useState([]);
+  const { setTitle, setTabRight } = useOutletContext();
 
   useEffect(() => {
     getConfig(id)
       .then((res) => {
-        setConfigDetails(res?.data[0]);
+        const data = res?.data[0];
+        setConfigDetails(data);
+        setTitle(
+          <BreadcrumbNav
```

Comment:
inline literals introduces unnecessary react dom reconciliations. 
Can we use memoized constants here.

Please handle all similar cases.